backend/src/v2/compiler/argocompiler/container.go

```diff
@@ -121,6 +122,24 @@ func (c *workflowCompiler) addContainerDriverTemplate() string {
 			Resources: driverResources,
 		},
 	}
+	// Add retry policy
+	if retryPolicy != nil {
+		maxRetryCount := intstr.FromInt(int(retryPolicy.GetMaxRetryCount()))
+		// During the converstion from float64 to int, the fraction is discarded (truncated towards 0).
+		// See https://go.dev/ref/spec#Conversions
+		backOffFactor := intstr.FromInt(int(retryPolicy.GetBackoffFactor()))
```

Comment:
We need to check the value of `retryPolicy.GetBackoffFactor()` and replace it with `2` if is set to `0.0`. Ref: https://github.com/kubeflow/pipelines/blob/master/api/v2alpha1/pipeline_spec.proto#L542-L544